### ISSUE TITLE
net/http/httputil: Question about new panic in ReverseProxy

### ISSUE THREAD

**Issue by VojtechVitek** [golang/go#28239]
### What version of Go are you using (`go version`)?
go version go1.11.1 darwin/amd64

### What did you do?
We use `httputil.ReverseProxy` in a HTTP Handler to proxy the S3 file download.

We started seeing `panic: net/http: abort Handler` messages in our logs/alert platform, since go1.11. 

The stacktrace leads to a new `panic(http.ErrAbortHandler) ` in `(*ReverseProxy).ServeHTTP()` method:

https://github.com/golang/go/blob/a0d6420d8be2ae7164797051ec74fa2a2df466a1/src/net/http/httputil/reverseproxy.go#L276-L287

### Question

Are we supposed to defer-recover from this new `panic()` from now on?

I feel this should be documented at https://golang.org/pkg/net/http/httputil/#ReverseProxy .. but it's not.

**Comment by VojtechVitek** [comment]
Also, can we get the original `err` from the `p.copyResponse()` somehow?

The panic obfuscates the original error with a generic `http.ErrAbortHandler` error.

**Comment by bradfitz** [comment]
Are you not using http.Server? The http.Server automatically handles ErrAbortHandler.

**Comment by bradfitz** [comment]
Also, #23643 which was included in Go 1.11 was supposed to even fix the case of not running under an http.Server.

So I'm really curious about your environment.

Do you have a minimal repro?

**Comment by VojtechVitek** [comment]
@bradfitz yes, we're using http.Server. The panic is handled and the program doesn't crash.

However, we use some HTTP middlewares that report all panics to Sentry and to our Logs. And now I'm curious what to do with this generic error.

Should we start filtering http.ErrAbortHandler out?
Can we get the original cause error somehow?

**Comment by bradfitz** [comment]
I don't know what your middleware wants to do, but here's the documentation for that panic value:

https://golang.org/pkg/net/http/#ErrAbortHandler

> ErrAbortHandler is a sentinel panic value to abort a handler. While any panic from ServeHTTP aborts the response to the client, panicking with ErrAbortHandler also suppresses logging of a stack trace to the server's error log.

You can modify your middleware as appropriate.

There is no way to get the original cause. Usually it's because the client went away (closed browser tab, etc) and the proxy copy thus failed.
